src/__tests__/ReactHardwareComponent-test.js

```diff
@@ -4,6 +4,11 @@ import ReactHardwareComponent from '../ReactHardwareComponent';
 import ReactHardwareReconcileTransaction from '../ReactHardwareReconcileTransaction';
 
 describe('ReactHardwareComponent', () => {
+  it('should warn if it gets an unknown type', () => {
+    // TODO: check so it is the correct error
+    expect(() => new ReactHardwareComponent({ type: 'foo' })).toThrow();
```

Comment:
nit: remove outter spacing in single line object literals (following React code style) `{type: 'foo'}`